qa/tasks/vstart_runner.py

```diff
@@ -773,6 +744,31 @@ def list_connections():
 
         self.gather_mount_info()
 
+    def _set_fuse_daemon_pid(self):
+        # NOTE: When a command <args> is launched with sudo, two processes are
+        # launched, one with sudo in <args> and other without. Make sure we
+        # get the PID of latter one.
+        i, _break = 0, False
+        while i < 5:
+            time.sleep(5)
+
+            lines = self.client_remote.run(
+                args=['ps', '-ef', Raw('|'), 'grep', 'ceph-fuse'],
```

Comment:
Test fails locally:

`python3 ../qa/tasks/vstart_runner.py tasks.cephfs.test_volume_client.TestVolumeClient.test_readonly_authorization`

```
2020-05-06 22:44:36,158.158 INFO:__main__:======================================================================
2020-05-06 22:44:36,158.158 INFO:__main__:ERROR: test_readonly_authorization (tasks.cephfs.test_volume_client.TestVolumeClient)
2020-05-06 22:44:36,158.158 INFO:__main__:----------------------------------------------------------------------
2020-05-06 22:44:36,158.158 INFO:__main__:Traceback (most recent call last):
2020-05-06 22:44:36,158.158 INFO:__main__:  File "/data/ceph/qa/tasks/cephfs/test_volume_client.py", line 19, in setUp
2020-05-06 22:44:36,158.158 INFO:__main__:    CephFSTestCase.setUp(self)
2020-05-06 22:44:36,159.159 INFO:__main__:  File "/data/ceph/qa/tasks/cephfs/cephfs_test_case.py", line 145, in setUp
2020-05-06 22:44:36,159.159 INFO:__main__:    self.mounts[i].mount_wait()
2020-05-06 22:44:36,159.159 INFO:__main__:  File "/data/ceph/qa/tasks/cephfs/mount.py", line 420, in mount_wait
2020-05-06 22:44:36,159.159 INFO:__main__:    mount_options=mount_options)
2020-05-06 22:44:36,159.159 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 720, in mount
2020-05-06 22:44:36,160.160 INFO:__main__:    self._set_fuse_daemon_pid()
2020-05-06 22:44:36,160.160 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 763, in _set_fuse_daemon_pid
2020-05-06 22:44:36,160.160 INFO:__main__:    stdout=StringIO()).stdout.getvalue().strip().split('\n')
2020-05-06 22:44:36,160.160 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 339, in run
2020-05-06 22:44:36,160.160 INFO:__main__:    return self._do_run(**kwargs)
2020-05-06 22:44:36,160.160 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 350, in _do_run
2020-05-06 22:44:36,161.161 INFO:__main__:    args = [a for a in args if a not in {
2020-05-06 22:44:36,161.161 INFO:__main__:  File "../qa/tasks/vstart_runner.py", line 351, in <listcomp>
2020-05-06 22:44:36,161.161 INFO:__main__:    'adjust-ulimits', 'ceph-coverage'}]
2020-05-06 22:44:36,161.161 INFO:__main__:TypeError: unhashable type: 'Raw'
2020-05-06 22:44:36,161.161 INFO:__main__:
```

Reply:
The issue is not related to this PR, IMHO. I've raised a ticket and PR for this -
https://tracker.ceph.com/issues/45446
https://github.com/ceph/ceph/pull/34962

Reply:
This still doesn't work:

```
2020-05-08 20:59:06,611.611 INFO:__main__:kill 
2020-05-08 20:59:06,612.612 INFO:__main__:kill: killing pid 6129 (['sudo', 'nsenter', '--net=/var/run/netns/ceph-ns--tmp-tmp2rvmh5r9-mnt.2', './bin/ceph-fuse', '--client_mountpoint=/volumes/grpid/volid', '-f', '--name', 'client.guest', '/tmp/tmp2rvmh5r9/mnt.2'])
```
The PID 6129 is still the sudo:

```
root      6129  0.0  0.0      0     0 pts/2    Z+   20:59   0:00 [sudo] <defunct>
root      6142  0.0  0.0   4372  1496 pts/2    S+   20:59   0:00 scl enable devtoolset-8  'nsenter' '--net=/var/run/netns/ceph-ns--tmp-tmp2
root      6145  0.0  0.0 113204  2708 pts/2    S+   20:59   0:00 /bin/bash /var/tmp/sclzEifIp
root      6163  0.2  0.2 1887248 39424 pts/2   Sl+  20:59   0:00 ./bin/ceph-fuse --client_mountpoint=/volumes/grpid/volid -f --name client.
root      6275  0.0  0.0      0     0 ?        I    20:59   0:00 [kworker/0:2-eve]
```